kernel/src/main/java/org/kframework/krun/KRun.java

```diff
@@ -77,13 +85,15 @@ public int run(CompiledDefinition compiledDef, KRunOptions options, Function<Mod
             program = parseConfigVars(options, compiledDef);
         }
 
-        program = new KTokenVariablesToTrueVariables()
-                .apply(compiledDef.kompiledDefinition.getModule(compiledDef.mainSyntaxModuleName()).get(), program);
+        // store initial configuration in single mutable reference so that we can make sure it can be garbage collected
```

Comment:
see here for what is going on with respect to the InitialConfiguration class. Basically I am trying to reduce the length of time that the initial configuration has to live in memory. it shows up a couple other places in this diff also, all aimed at the same thing.